go/libkb/merkle_client.go

```diff
@@ -394,19 +395,36 @@ func GetNodeHashVoid(w *jsonw.Wrapper, nhp *NodeHash, errp *error) {
 	}
 }
 
-func computeSetBitsBigEndian(x int) []int {
-	var ret []int
-	for bit := 1; bit <= x; bit *= 2 {
+func computeSetBitsBigEndian(x uint) []uint {
+	if x == 0 {
+		return nil
+	} else if x == 1 {
+		return []uint{1}
+	}
+	// Allocate maximum array size necessary
+	high := int(math.Ceil(math.Log2(float64(x))))
+	ret := make([]uint, high)
```

Comment:
it might be easier to `ret = make([]uint, 0, high)`, and then just append?

Reply:
ah, I didn't know about this. Thanks!